realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -897,24 +897,139 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeNotEqual__J_3JLja
     TableQuery_StringPredicate(env, nativeQueryPtr, columnIndexes, value, caseSensitive, StringNotEqual);
 }
 
-JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith(
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith__J_3JLjava_lang_String_2Z(
```

Comment:
I just used the [regenerate-jni-headers.sh](https://github.com/realm/realm-java/blob/master/realm/realm-jni/generate-jni-headers.sh) script which is at `realm/realm-jni/`, but yes, it looks like that's the case.

For what it's worth, those generated headers (for the `String` methods) didn't change until I added the equivalent ones for `byte[]`. I assume that at that point javah had to output more explicit header names to help distinguish them for some reason. It seems that the `String` headers for `nativeEqualTo` and `nativeNotEqualTo` already had that extra suffix, so I assume this is a standard thing for javah output.